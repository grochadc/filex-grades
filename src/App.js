import React, { useState } from "react";
import styled, { createGlobalStyle } from "styled-components";
import Form from "./components/Form";
import Grades from "./components/Grades";
import NameChooser from "./components/NameChooser";
import grades from "./data/grades.json";

const GlobalStyle = createGlobalStyle`
body {
  color: white;
  background-color:#282c34;
}
`;

const Title = styled.h1`
  text-align: center;
`;

function App() {
  const [code, setCode] = useState();
  return (
    <div>
      <GlobalStyle />
      <Title>Teacher Gonzalo's Final Grades</Title>
      <Form setCode={setCode} />
      {code ? (
        code === "EXTERNO" ? (
          <NameChooser
            names={["Julia", "Karen", "Leticia"]}
            setCode={setCode}
          />
        ) : (
          <Grades code={code} />
        )
      ) : null}
    </div>
  );
}

export default App;
