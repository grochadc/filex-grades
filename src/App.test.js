import React from "react";
import App from "./App";
import { render, cleanup, fireEvent, getByTestId } from "react-testing-library";

afterEach(cleanup);
describe("App", () => {
  it("renders without crashing", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
