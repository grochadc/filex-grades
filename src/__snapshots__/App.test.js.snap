// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <div>
    <section
      class="sc-htoDjs blDzrv"
    >
      <h1
        class="sc-gzVnrw hDwEie"
      >
        Teacher Gonzalo's Final Grades
      </h1>
      <div
        class="sc-bdVaJa khUSez"
      >
        Code:
         
        <input
          type="text"
          value=""
        />
         
        <button>
          Send
        </button>
        <p>
          Eres externo y no tienes codigo? Click
           
          <button
            class="sc-bwzfXH jAzBCL"
            data-testid="externos-link"
          >
            aqui
          </button>
          .
        </p>
      </div>
    </section>
    <section
      class="sc-htoDjs bVebbx"
    />
  </div>
</div>
`;
